gtsfm/data_association/dsf_tracks_estimator.py

```diff
@@ -46,44 +46,28 @@ def run(self, matches_dict: Dict[Tuple[int, int], np.ndarray], keypoints_list: L
         if not dims_valid:
             raise Exception("Dimensions for Keypoint coordinates incorrect. Array needs to be 2D")
 
-        # Generate the DSF to form tracks
-        dsf = gtsam.DSFMapIndexPair()
-        track_2d_list = []
-        # for DSF finally
-        # measurement_idxs represented by ks
+        # converting python dict into gtsam.MatchIndicesMap
+        matches_map = gtsam.MatchIndicesMap()
         for (i1, i2), k_pairs in matches_dict.items():
-            for (k1, k2) in k_pairs:
-                dsf.merge(gtsam.IndexPair(i1, k1), gtsam.IndexPair(i2, k2))
-
-        key_set = dsf.sets()
-
-        erroneous_track_count = 0
-        # create a landmark map: a list of tracks
-        # Each track is represented as a list of (camera_idx, measurements)
-        for set_id in key_set:
-            index_pair_set = key_set[set_id]  # key_set is a wrapped C++ map, so this unusual syntax is required
-
-            # Initialize track from measurements
-            track_measurements = []
-            for index_pair in gtsam.IndexPairSetAsArray(index_pair_set):
-                # camera_idx is represented by i
-                # measurement_idx is represented by k
-                i = index_pair.i()
-                k = index_pair.j()
-                # add measurement in this track
-                track_measurements += [SfmMeasurement(i, keypoints_list[i].coordinates[k])]
-
-            track_2d = SfmTrack2d(track_measurements)
+            matches_map[gtsam.IndexPair(i1, i2)] = k_pairs
+        
+        # converting gtsfm Keypoints into gtsam Keypoints
+        keypoints_vector = gtsam.KeypointsVector()
+        for keypoint in keypoints_list:
+            keypoints_vector.append(
+                gtsam.gtsfm.Keypoints(keypoint.coordinates)
+            )
+        
+        tracks = gtsam.gtsfm.tracksFromPairwiseMatches(
```

Comment:
Thanks @stepanyanhayk! This looks good. Could you add a simple test for this DSF tracks estimator, using the example keypoints + matches from this test? https://github.com/borglab/gtsam/blob/develop/python/gtsam/tests/test_DsfTrackGenerator.py#L22-L35

I believe you can use it verbatim.

Reply:
Thanks for reviewing, John. Added a unittest for DSF track estimation in a new commit.

Reply:
Marking as resolved, thanks!